src/Symfony/Component/PropertyInfo/Tests/TypeTest.php

```diff
@@ -37,6 +46,26 @@ public function testConstruct()
         $this->assertEquals(Type::BUILTIN_TYPE_STRING, $collectionValueType->getBuiltinType());
     }
 
+    public function testConstruct()
```

Comment:
Would it be possible to add support for unions outside of the arrays as well? e.g. `int|string $foo`

Reply:
Actually we do parse such strings in the extractor, so I think we should keep array structure for the Type representation and for string representation it should be handled by extractors.

Reply:
Hm ok will have to check how it's done because I'm pretty sure this is currently not working in API-Platform

Reply:
Actually, this is not working for collection key & value types since we have no extractor managing this behavior, look the issue linked to this PR for more details.